### DISCUSSION TITLE
[blocks]: Multistep form with validation

### DISCUSSION THREAD

**Discussion by m4recek** [shadcn-ui/ui#3796] [Blocks Request]
### Description

Form with multiple steps with recap & confirmation at the end (think ordering on an eshop).

**Comment by anishshobithps** [comment]
@m4recek there is a [PR for Stepper function](https://github.com/shadcn-ui/ui/pull/318),

Also there are [some great examples](https://github.com/damianricobelli/shadcn-stepper/tree/main/examples) you can find by the author of the PR

**Reply by Steveb599** [reply]
but where does he use the validation?

**Reply by anishshobithps** [reply]
There is a [example with a form, which uses zod](https://github.com/damianricobelli/shadcn-stepper/blob/main/examples/stepper-form.tsx) are you looking for this @Steveb599 ?

**Reply by Steveb599** [reply]
Yes thank you I'll try it.

**Reply by Steveb599** [reply]
tried it but not quite exactly because when clicking on prev the data disappears (probably needs useContext)

**Reply by anishshobithps** [reply]
I think you can put the comment on the open PR so while it is, at least the Author can document it as well.
@Steveb599
